### DISCUSSION TITLE
Should modal overlays remain open when another overlay is added and when they're not an ancestor of a click event?

### DISCUSSION THREAD

**Discussion by lehelen19** [adobe/spectrum-web-components#4463] [Q&A]
1. When we have a modal overlay open, adding another auto, modal, or page overlay will close it: https://github.com/adobe/spectrum-web-components/blob/main/packages/overlay/src/OverlayStack.ts#L176
- Example suggested change: https://github.com/lehelen19/spectrum-web-components/blob/b525bf565e31cfb1be49a11ff65fd25c18f5bcfd/packages/overlay/src/OverlayStack.ts#L175

2. When a modal overlay is not an ancestor of a click event, it will close: https://github.com/adobe/spectrum-web-components/blob/main/packages/overlay/src/OverlayStack.ts#L86-L89
- Example suggested change: https://github.com/lehelen19/spectrum-web-components/blob/b525bf565e31cfb1be49a11ff65fd25c18f5bcfd/packages/overlay/src/OverlayStack.ts#L90

This behavior is technically not consistent with how modal overlays functioned before the Overlay v2 upgrade, so I wanted to recommend preventing these modal overlays from closing on an overlay addition and when another overlay closes while on top due to a pointer interaction. I'd like to hear more about whether these changes (or something similar with a different implementation!) would prove beneficial in general. Thank you!

**Comment by Westbrook** [comment]
For reference: https://github.com/adobe/spectrum-web-components/issues/4193#issuecomment-2008132306
